presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/IndexJoinOptimizer.java

```diff
@@ -551,6 +557,15 @@ public void markSuccess()
                 checkState(node.getLookupSymbols().equals(lookupSymbols), "lookupSymbols must be the same as IndexSource lookup symbols");
                 return lookupSymbols.stream().collect(toImmutableMap(identity(), identity()));
             }
+
+            private Symbol extractSymbol(RowExpression expression)
+            {
+                if (expression instanceof VariableReferenceExpression) {
```

Comment:
Add a comment to explain why an optimizer needs to handle `RowExpression` (for `ValidateDependenciesChecker`).